stardog/admin.py

```diff
@@ -956,3 +1008,107 @@ def available(self):
 
     def __repr__(self):
         return self.name
+
+
+class DataSource(object):
+    """Initializes a DataSource
+
+    See Also:
+        https://docs.stardog.com/virtual-graphs/virtual-sources
+    """
+
+    def __init__(self, name, client):
+        """Initializes a DataSource.
+
+        Use :meth:`stardog.admin.Admin.data_source`,
+        :meth:`stardog.admin.Admin.data_sources`, or
+        :meth:`stardog.admin.Admin.new_data_source` instead of
+        constructing manually.
+        """
+
+        self.data_source_name = name
+        self.path = '/admin/data_sources/{}'.format(name)
+        self.client = client
+
+    @property
+    def name(self):
+        """The name of the data source.
+        """
+        return self.data_source_name
+
+    def available(self):
+        """Checks if the data source is available.
+
+        Returns:
+          bool: Availability state
+        """
+        r = self.client.get(self.path + '/available')
+        return bool(r.json())
```

Comment:
this looks suspicious.  Wont this return a dictionary?  The doc says that this endpoint returns plain text